dataflow/src/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -142,7 +149,29 @@ protected S runAnalysisFor(Node node, boolean before) {
         if (transferInput == null) {
             return null;
         }
-        return runAnalysisFor(node, before, transferInput);
+        return runAnalysisForWithCacheLookupMap(node, before, transferInput, cacheLookupMap);
+    }
+
+    public static <A extends AbstractValue<A>, S extends Store<S>>
```

Comment:
Please add javadoc to this public method, in particular highlighting how to choose between the different `runAnalysisFor` methods.